### PR TITLE
feat: Fetch acme directory root path from the team settings WPB-5706

### PR DESCRIPTION
**Pull request by KaterinaWire**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-5706" title="WPB-5706" target="_blank">StoryWPB-5706</a>  [iOS] Fetch acme directory root path from the team settings
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [x] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?
In this PR I added support for the new `mlsE2EId` feature from the team settings. It has the structure:
```
mlsE2EId": {
    "config": {
      "acmeDiscoveryUrl": "https://example.com",
      "verificationExpiration": 9223372036854776000
    },
    "lockStatus": "locked",
    "status": "enabled",
    "ttl": "unlimited"
  }
```

And pass acmeDiscoveryUrl from the `mlsE2EId` feature to the `AcmeAPI`.
----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: wire-ios-data-model/Source/Model/FeatureConfig/E2EId/Feature.E2EId.swift [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import Foundation
+
+public extension Feature {
+
+    struct E2EId: Codable {
```
Comment (netbe) [inline wire-ios-data-model/Source/Model/FeatureConfig/E2EId/Feature.E2EId.swift]: why is it called E2EI**d** ? what does the d stand for?

File: wire-ios-data-model/Source/Model/FeatureConfig/Feature.swift [hunk 2]
```diff
@@ -218,6 +219,9 @@ public class Feature: ZMManagedObject {
 
         case .conferenceCalling, .fileSharing, .conversationGuestLinks, .classifiedDomains, .digitalSignature, .mls:
             break
+        case .e2ei:
+            /// TODO: in the next PR
```
Comment (netbe) [inline wire-ios-data-model/Source/Model/FeatureConfig/Feature.swift:223]: do you have a ticket id to mention here?

**Review by netbe — COMMENTED** [review]
LGTM, just some questions on naming

File: wire-ios-data-model/Source/Model/FeatureConfig/Feature.swift (hunk 2 above)
Comment (KaterinaWire) [reply wire-ios-data-model/Source/Model/FeatureConfig/Feature.swift:223]: Yes, I have a ticket. Added to the comment.

File: wire-ios-data-model/Source/Model/FeatureConfig/E2EId/Feature.E2EId.swift (hunk 1 above)
Comment (KaterinaWire) [reply wire-ios-data-model/Source/Model/FeatureConfig/E2EId/Feature.E2EId.swift]: This's a feature from the TM, and as a part of the config response from BE we receive:
```
...
"mlsE2EId": {
    "config": {
      "acmeDiscoveryUrl": "https://example.com",
      "verificationExpiration": 9223372036854776000
    },
    "lockStatus": "locked",
    "status": "enabled",
    "ttl": "unlimited"
  },
  ...
```
That's why I used E2EI**d**. But you're right, we usually only use E2EI. I changed this in the last commit.

**Review by netbe — APPROVED** [review]

**Review by David-Henner — APPROVED** [review]

**Pull request merged**